src/lints/tuple_struct_to_plain_struct.ron

```diff
@@ -0,0 +1,62 @@
+SemverQuery(
+    id: "tuple_struct_to_plain_struct",
+    human_readable_name: "tuple struct changed to plain struct",
+    description: "A struct changed from a tuple struct to plain struct",
+    reference: Some(r#"
+This is breaking even if the fields have the same names, because tuple struct patterns won't work.
+Source: Rust for Rustaceans, Chapter 3, "Type Modifications", page 51
+"#),
+    required_update: Major,
+    reference_link: None,
+    query: r#"
+    {
+        CrateDiff {
+            baseline {
+                item {
+                    ... on Struct {
+                        visibility_limit @filter(op: "=", value: ["$public"]) @output
+                        name @tag @output
+                        attrs @filter(op: "not_contains", value: ["$non_exhaustive"])
+                        struct_type @filter(op: "=", value: ["$tuple"])
+
+                        # Filter out structs that have private fields
+                        field @fold @transform(op: "count") @filter(op: "=", value: ["$zero"]) {
+                            visibility: visibility_limit @filter(op: "!=", value: ["$public"])
+                        }
+
+                        importable_path {
+                            path @tag @output
+                        }
+                    }
+                }
+            }
+            current {
+                item {
+                    ... on Struct {
```

Comment:
If the struct didn't use to be `#[non_exhaustive]` but now became `#[non_exhaustive]`, that's a different breaking change that we perhaps shouldn't report here: non-exhaustive structs cannot be constructed with a literal nor pattern-matched, and the author of that code has explicitly marked the struct as such.

Instead, we'd report that under the "struct used to be exhaustive but now is non-exhaustive" lint.

We probably want an `attr @filter(...)` here and a test case to check for this.